samples/drm/app/main.js

```diff
@@ -0,0 +1,374 @@
+'use strict';
+
+angular.module('DashSourcesService', ['ngResource']).
+    factory('Sources', function($resource){
+        return $resource('app/sources.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashNotesService', ['ngResource']).
+    factory('Notes', function($resource){
+        return $resource('app/notes.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashContributorsService', ['ngResource']).
+    factory('Contributors', function($resource){
+        return $resource('app/contributors.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashPlayerLibrariesService', ['ngResource']).
+    factory('PlayerLibraries', function($resource){
+        return $resource('app/player_libraries.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+angular.module('DashShowcaseLibrariesService', ['ngResource']).
+    factory('ShowcaseLibraries', function($resource){
+        return $resource('app/showcase_libraries.json', {}, {
+            query: {method:'GET', isArray:false}
+        });
+    });
+
+var app = angular.module('DashPlayer', [
+    'DashSourcesService',
+    'DashNotesService',
+    'DashContributorsService',
+    'DashPlayerLibrariesService',
+    'DashShowcaseLibrariesService'
+]);
+
+app.controller('DashController', function($scope, Sources, Notes, Contributors, PlayerLibraries, ShowcaseLibraries) {
+    var player,
+        video,
+        context;
+
+    $scope.drmData = [];
+
+    $scope.safeApply = function(fn) {
+        var phase = this.$root.$$phase;
+        if(phase == '$apply' || phase == '$digest')
+            this.$eval(fn);
+        else
+            this.$apply(fn);
+    };
+
+    ////////////////////////////////////////
+    //
+    // Error Handling
+    //
+    ////////////////////////////////////////
+
+    function onError(e) {
+
+    }
+
+    ////////////////////////////////////////
+    //
+    // Player Setup
+    //
+    ////////////////////////////////////////
+
+    video = document.querySelector(".dash-video-player video");
+    context = new Dash.di.DashContext();
+    player = new MediaPlayer(context);
+    $scope.version = player.getVersion();
+
+    player.startup();
+    player.addEventListener(MediaPlayer.events.ERROR, onError.bind(this));
+
+    player.attachView(video);
+    player.setAutoPlay(true);
+
+    var system = player.getSystem(),
```

Comment:
@greg80303 @KozhinM .  I am not fully up on the use case but I think I understand.  Pre fetching license info before playback starts correct?  I tend to agree with @KozhinM here that it seems like we should have API at media player level OR we have a module that can be used externally and internal to achieve the needs for the use case.  Should we discuss more on a call and try to come to a good architecture?